docs/A-Introduction/computers.md

```diff
@@ -18,15 +22,17 @@ Computers are available in many flavours: mobile devices, smart phones, laptops,
 
 We refer to the software with which a user operates their device as application software. Application software consists of one or more programs. Each program is a full set of instructions that performs a well-defined task on the host device. Programmers code these instructions in a programming language.
 
-The instructions that programmers code represent algorithms. An algorithm is a step-by-step procedure that describes how to achieve a specified task. Examples include searching, sorting and mixing. Application programmers study different algorithms and create their own if required. They find the code for some algorithms in libraries and write the code for their own algorithms.
+The instructions that programmers code represent [algorithms](/F-Refinements/algorithms "Algorithms"). An algorithm is a step-by-step procedure that describes how to achieve a specified task. Examples include searching, sorting and mixing. Application programmers study different algorithms and create their own if required. They find the code for some algorithms in [libraries](/D-Modularity/library-functions "Library Functions") and write the code for their own algorithms.
 
 This introductory chapter describes the major components of a modern computer, components to which programmers often refer. Subsequent chapters show how to write programs to use the principal features of these components efficiently.
 
 ## Hardware
 
 The figure below illustrates the relation of the programmer to a user of a software application. The boxes identify the principal elements for any programmer. The green boxes identify the elements of concern to the user.
 
-![](https://ict.senecacollege.ca//~ipc144/pages/images/programmer.png)
+<p align="center">
+    <img src="https://ict.senecacollege.ca//~ipc144/pages/images/programmer.png" alt="Diagram of User input goes to Hardware(containing Memory) and User output comes out. While Programmer provides Instructions to the Hardware."/>
+</p>
```

Comment:
I believe that it will be better if we use as less HTML syntax as possible. Therefore, for all the images, I suggest that you convert them to markdown. For example, the above image with alt message in markdown would look like this:

`[Diagram of User input goes to Hardware(containing Memory) and User output comes out. While Programmer provides Instructions to the Hardware.](https://ict.senecacollege.ca//~ipc144/pages/images/programmer.png)`

In addition to that, I recommend you to shorten your alt messages, because they seem a bit too long. Also, you don't have to mention that it is "diagram of"/"picture of" in your alt messages, because the browser will already know what kind of data it is.

Don't worry about centering images for now, because after PR #46 gets merged, you can center the images like how #63 explains.

Reply:
What do you recommend to use as alt text instead? I'm having trouble thinking of how to describe it, I figured that saying it is a diagram would be ok because that is more specific than saying picture, but I understand what you mean.

Reply:
Maybe something like "Major overview of a generic program"?  @joshuali7536